### PR TITLE
Add cache-busting script for stylesheet

### PR DESCRIPTION
**Pull request by dpblume**
Adds a cache-busting script to append a "version" to the query parameter that imports our stylesheet.

This ensures that the updated `outputs.css` file is loaded whenever a user opens a webpage.

Example when inspecting the rendered HTML in Chrome:

image

### CODE REVIEW

**Review by nathanaelrobinson — APPROVED** [review]
One small nit I want to verify before you merge -- I think as I read this every single time the page loads a new cachebuster will fire -- which means that the cache will be invalidated every single time someone reloads the page. Would slow down user access when we're not making updates.

Maybe we want to use a github commit hash on main to ensure that's not the case?

You see where I'm going with this? If every single time someone hits the page they get a new "v=123232" then it'll will ask for the css file 100% of the time, even if they reload it 2x in a 1 minute period.

We really want to make this happen only if there's a new commit -- so maybe and I'm just being crazy here -- but we have a preload script or something on server startup time that finds the commit hash and injects that as a variable so that the CSS is static for each deploy, but won't change aside from that?

**Comment by dpblume** [comment]
Yeah that's a great callout. I'll see if I can get it working with the commit hash, because that seems like a much better way to handle this

**Comment by nathanaelrobinson** [comment]
> Yeah that's a great callout. I'll see if I can get it working with the commit hash, because that seems like a much better way to handle this

Yeah sor

File: views.go [hunk 1]
```diff
@@ -88,7 +89,12 @@ func aboutHandler(w http.ResponseWriter, req *http.Request) {
...
 // Services page for all /services url requests
 func servicesHandler(w http.ResponseWriter, req *http.Request) {
-	http.ServeFile(w, req, "./templates/services.html")
+	var githash = os.Getenv("gitHash")
+	servicesTemplate, _ := template.ParseFiles("./templates/services.html")
+	err := servicesTemplate.Execute(w, githash)
+	if err != nil {
+		log.Fatal(err)
+	}
 }
```
Comment (dpblume) [inline views.go:98]: @nathanaelrobinson would you mind taking a look at this part? I'm definitely doing it incorrectly and throwing an error

Comment (nathanaelrobinson) [reply views.go:98]: Yeah I got you I can add a commit

File: views.go [hunk 2]
```diff
@@ -166,16 +197,26 @@ func blogTagHandler(w http.ResponseWriter, req *http.Request) {
...
 	}
+	type PageData struct {
+		Posts   []Post
+		GitHash string
+		Subject string
+	}
+	pageData := PageData{
+		Posts:   sortedBlogs,
+		GitHash: a.GitHash,
+		Subject: tag,
```
Comment (nathanaelrobinson) [inline views.go:208]: I have included a little easter egg for you here. This is the handler for sending someone all of the blogs based on a filter by tags in the url. I haven't done it yet but you can easily update the template to capture this `Subject` variable and display it on the page. The one thing to keep in mind is that it's the same template used for the blog landing page so you'll need a fallback option if the Subject variable isn't passed to the template -- worth taking a look at the `http/template` documentation for Go if you're interested in that.

**Pull request merged**